bundles/org.openhab.binding.vdr/src/main/java/org/openhab/binding/vdr/internal/VDRHandler.java

```diff
@@ -0,0 +1,357 @@
+                            entry = con.getEpgEvent(SVDRPEpgEvent.TYPE.NEXT);
+                            result = new StringType(entry.getSubtitle());
+                            break;
+                        case "nextEventDuration":
+                            entry = con.getEpgEvent(SVDRPEpgEvent.TYPE.NEXT);
+                            result = new DecimalType(entry.getDuration());
+                            break;
+                        case "nextEventBegin":
+                            entry = con.getEpgEvent(SVDRPEpgEvent.TYPE.NEXT);
+                            result = new DateTimeType(LocalDateTime
+                                    .ofInstant(entry.getBegin().toInstant(), TimeZone.getDefault().toZoneId())
+                                    .atZone(TimeZone.getDefault().toZoneId()));
+                            break;
+                        case "nextEventEnd":
+                            entry = con.getEpgEvent(SVDRPEpgEvent.TYPE.NEXT);
+                            result = new DateTimeType(LocalDateTime
+                                    .ofInstant(entry.getEnd().toInstant(), TimeZone.getDefault().toZoneId())
+                                    .atZone(TimeZone.getDefault().toZoneId()));
+                            break;
+
+                    }
+                    updateState(channelUID, result);
+
+                } catch (SVDRPParseResponseException e) {
+                    logger.trace("VDR Refresh for Thing {}, ChannelUID {}, Parse Response failed. Message: {}",
+                            this.getThing().getUID(), channelUID, e.getMessage());
+                    updateState(channelUID, UnDefType.UNDEF);
+                } catch (SVDRPConnectionException e) {
+                    logger.debug("VDR Refresh for Thing {}, ChannelUID {}, Connection failed. Message: {}",
+                            this.getThing().getUID(), channelUID, e.getMessage());
+                }
+
+            });
+
+        } catch (SVDRPException ce) {
+            if (thing.getStatus() == ThingStatus.ONLINE) {
+                // also update power channel when thing is offline before setting it offline
+                thing.getChannels().stream().map(c -> c.getUID()).filter(this::isLinked).forEach(channelUID -> {
+                    try {
+                        if ("power".equals(channelUID.getIdWithoutGroup())) {
+                            updateState(channelUID, OnOffType.OFF);
+                        }
+                    } catch (Exception e) {
+                        logger.warn("VDR Refresh for Thing {}, ChannelUID {} failed. ErrorMessage: {}",
+                                this.getThing().getUID(), channelUID, e.getMessage());
+                    }
+                });
+            }
+            updateStatus(ThingStatus.OFFLINE);
+
+        } catch (Exception ex) {
+            logger.error("There is something wrong with your thing '{}', please check or recreate it: {}",
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.